test/test_triage.py

```diff
@@ -0,0 +1,450 @@
+import sys
+import json
+
+try:
+    import unittest.mock as umock
+except ImportError:
+    import mock as umock
+import unittest
+import logging
+import tempfile
+import shutil
+import glob
+import os
+
+# hysds.celery searches for configuration on import. So we need to make sure we
+# mock it out before the first time it is imported
+sys.modules["hysds.celery"] = umock.MagicMock()
+logging.basicConfig()
+
+
+class TestTriage(unittest.TestCase):
+    def setUp(self):
+        self.job_dir = tempfile.mkdtemp(prefix="job-")
+        logging.info("self.job_dir: {}".format(self.job_dir))
+
+    def tearDown(self):
+        umock.patch.stopall()
+        shutil.rmtree(self.job_dir)
+
+    def test_triage_job_succeeded(self):
+        import hysds.triage
+
+        job = {"job_info": {"status": 0}}
+        self.assertTrue(hysds.triage.triage(job, None))
+
+    def test_triage_disabled(self):
+        import hysds.triage
+
+        job = {"job_info": {"status": 1}}
+        job_context = {"_triage_disabled": True}
+        self.assertTrue(hysds.triage.triage(job, job_context))
+
+    def test_triage_default_triage_id(self):
+        import hysds.triage
+
+        # Test case data
+        _id = "da9be25e-e281-4d3c-a7d8-e3c0c8342972"
+        triage_name = f"triaged_job-boogaloo_task-{_id}"
+        job = {
+            "task_id": _id,
+            "job_info": {
+                "id": "boogaloo",
+                "status": 1,
+                "job_dir": self.job_dir,
+                "time_start": "0001-01-01T00:00:00.000Z",
+                "context_file": "electric",
+                "datasets_cfg_file": "more/configuration",
+                "metrics": {"product_provenance": dict(), "products_staged": list()},
+            },
+        }
+
+        job_context = {"_triage_disabled": False}
+
+        # Mocked data
+        open_mock = umock.patch("hysds.triage.open", umock.mock_open()).start()
+        publish_dataset_mock = umock.patch("hysds.dataset_ingest.publish_dataset").start()
+        publish_dataset_mock.return_value = {}
+
+        # Expectations
+        expected_triage_dataset_filename = self.job_dir + f"/{triage_name}/{triage_name}.dataset.json"
+        expected_triage_met_filename = self.job_dir + f"/{triage_name}/{triage_name}.met.json"
+        expected_triage_json_filename = self.job_dir + "/_triaged.json"
+
+        # Test execution
+        result = hysds.triage.triage(job, job_context)
+
+        # Assertions
+        self.assertTrue(result)
+
+        open_mock.assert_any_call(expected_triage_dataset_filename, umock.ANY)
+        open_mock.assert_any_call(expected_triage_met_filename, umock.ANY)
+        open_mock.assert_any_call(expected_triage_json_filename, umock.ANY)
+
+    @umock.patch("hysds.dataset_ingest.publish_dataset")
```

Comment:
@mcayanan i had to move the `hysds.dataset_ingest.publish_dataset` mock to a decorator to get it to work, it was raising some error with file not found

are you okay with this solution? any ideas on how i can get it to work without using this decorator

Reply:
Per our tag up, let's see what @pymonger says. I'm not an expert on the mock stuff and I'm not sure why moving it out as a decorator worked. It was weird though that when the unit test was left as-is (the original implementation), that it looked like it was actually calling the `publish_dataset` function. It should have "mocked" it instead. My guess is that moving stuff around clobbered something to where basically the test doesn't know that this is function should be mocked.

Reply:
Sorry just catching up on things. What was the error @DustinKLo?

Reply:
it wasnt mocking one of the functions properly and says `electric` not found

Reply:
@DustinKLo: the original test function shows:
```
        # Mocked data
        open_mock = umock.patch("hysds.utils.open", umock.mock_open()).start()
        makedirs_mock = umock.patch("os.makedirs").start()
        shutil_copy_mock = umock.patch("shutil.copy").start()
        publish_dataset_mock = umock.patch("hysds.utils.publish_dataset").start()
        publish_dataset_mock.return_value = {}
```
Perhaps try adding back the mock for `os.makedirs` and `shutil.copy` and removing the decorator?
```
       # Mocked data
        open_mock = umock.patch("hysds.triage.open", umock.mock_open()).start()
       makedirs_mock = umock.patch("os.makedirs").start()
        shutil_copy_mock = umock.patch("shutil.copy").start()
        publish_dataset_mock = umock.patch("hysds.dataset_ingest.publish_dataset").start()
        publish_dataset_mock.return_value = {}
```